src/service.rs

```diff
@@ -470,6 +466,26 @@ impl Service {
                         self.send_ping(enr, None);
                     }
                 }
+                connectivity_timeout = self.connectivity_state.poll() => {
+                    match connectivity_timeout {
+                        TimerFailure::V4 => {
+                            // We have not received enough incoming connections in the required
+                            // time. Remove our ENR advertisement.
+                            info!(ip_version="v4", next_attempt_in=%DURATION_UNTIL_NEXT_CONNECTIVITY_ATTEMPT.as_secs(), "UDP Socket removed from ENR");
+                            if let Err(error) = self.local_enr.write().remove_udp_socket(&self.enr_key.read()) {
+                                error!(?error, "Failed to update the ENR");
+                            }
+                        }
+                        TimerFailure::V6 => {
+                            // We have not received enough incoming connections in the required
+                            // time. Remove our ENR advertisement.
+                            info!(ip_version="v6", next_attempt_in=%DURATION_UNTIL_NEXT_CONNECTIVITY_ATTEMPT.as_secs(), "UDP Socket removed from ENR");
+                            if let Err(error) = self.local_enr.write().remove_udp6_socket(&self.enr_key.read()) {
+                                error!(?error, "Failed to update the ENR");
+                            }
+                        }
+                    }
```

Comment:
Should we ping_connected_peers to advertise the updated ENR?

Reply:
Yep, nice catch!